[ImMemView] Editable Memory

gooosedev opened the pull request:
This PR enables editing of the memory  from the hexview,
as of now, only editing in the hex part of the view is possible, 
if the cursor is inside the ascii column the edit will be ignored
there is a toggle added in the settings to allow/deny editing of the memory.
![editMemory](https://github.com/user-attachments/assets/ea89fbba-b3ab-4800-93ec-86be078a408b)

gooosedev committed:
Editable Memory using the ImMemView

hrydgard commented on UI/ImDebugger/ImMemView.cpp:
```diff
@@ -319,9 +319,43 @@ void ImMemView::ProcessKeyboardShortcuts(bool focused) {
 			ScrollWindow(-visibleRows_, GotoModeFromModifiers(false));
 		}
+
+		if(editableMemory_){
```
try to follow the style of the surrounding code - for example, we use space after `if` and before `{`, and around operators.

hrydgard commented on UI/ImDebugger/ImMemView.cpp:
```diff
@@ -319,9 +319,43 @@ void ImMemView::ProcessKeyboardShortcuts(bool focused) {
+	uint8_t og= Memory::ReadUnchecked_U8(curAddress_);
+	u8 out;
+	if(selectedNibble_){
+		out = og ^((og^newNibble)&0xf);
```
wouldn't this be simpler as just `(og & 0xF0) | newNibble` ? 

similarly below, `(og & 0x0F) | (newNibble << 4)`

gooosedev committed:
ImMemView:
code-style cleanup
use simple bit operations to edit nibbles in memory edit mode
